### ISSUE TITLE
Allow setting a default destination

### ISSUE THREAD

**Issue by jeromedalbert** [basecamp/kamal#1025]
I have a `config/deploy.yml` with common configuration options, and a `config/deploy.production.yml`. I could very well have a `config/deploy.staging.yml` or what have you.

Whenever I deploy, I find it bothersome to always type `kamal deploy -d production`.

Would it make sense to add an option to somehow configure a default destination? Maybe something `default_destination: production` in `config/deploy.yml`, or somewhere else.

In the meantime, I have the following hacky workaround in my `bin/kamal` Rails binstub:

```ruby
...

# Make production the default Kamal destination
require 'thor'
destination =
  Thor::Options.new(_: Thor::Option.new(:destination, { aliases: '-d' })).parse(ARGV)['destination']
if destination.nil?
  destination = 'production'
  ARGV.push('-d', 'production')
end

load Gem.bin_path('kamal', 'kamal')
```

**Comment by dhh** [comment]
What's stopping you from having production being just deploy.yml and then overwriting what's needed in deploy.staging.yml?

**Comment by jeromedalbert** [comment]
Nothing, maybe I am just set in my ways!

I guess I like explicit destination names (`production`/`staging`/etc) that map to explicit filenames (`config/deploy.production.yml`/`config/deploy.staging.yml`/etc), and having `deploy.yml` as a destination-less common config file. Similar idea as what is already done with `.kamal/secrets-common` being a common secret file.

A yaml setup like this makes it obvious to anyone what file deploys to what destination, and defining a default destination would just be a nice-to-have on top of that.

Feel free to close if that feels niche, I am fine either way since I am using the workaround mentioned above!

**Comment by dhh** [comment]
Yeah, we've been exploring something similar at 37s but ultimately ended up on the idea that the default should just be production. That that's what "deploy" means without qualification.

**Comment by jeromedalbert** [comment]
That makes sense, I'm going to give your concept of "`deploy.yml` == production" a try in order to be closer to stock Kamal.

And I can add a comment at the top of `deploy.yml` to make it explicit that the file is for deploying to production. This way things are still explicit!

**Comment by jeromedalbert** [comment]
On second thought I prefer my default destination to be `staging` so I don't accidentally push to production. This way I manually need to type `-d production` to deploy to prod, and I actually want this extra friction because production deploys are more dangerous.

This only makes sense for manual deploys, it's a moot point if CI does the deploy for you.

So with that setup, `config/deploy.yml` would be deploying to staging, and I don't know if this name makes sense. Maybe it still does as "deploy" can mean "the first deploy should always be to staging". I think I'm keeping my workaround for now, but I'll probably go back and forth on the idea.

**Comment by jmschneider** [comment]
I was also hoping for a default destination option. I was hoping to set the destination from the current git branch in order to tie different branches to different environments.
